keps/sig-apps/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,453 @@
+  At a high level, this usually will be in the form of "high percentile of SLI
+  per day <= X". It's impossible to provide comprehensive guidance, but at the very
+  high level (needs more precise definitions) those may be things like:
+  - per-day percentage of API calls finishing with 5XX errors <= 1%
+  - 99% percentile over day of absolute value from (job creation time minus expected
+    job creation time) for cron job <= 10%
+  - 99,9% of /health requests per day finish with 200 code
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+  Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
+  implementation difficulties, etc.).
+
+### Dependencies
+
+_TBD upon graduation to beta._
+
+* **Does this feature depend on any specific services running in the cluster?**
+  Think about both cluster-level services (e.g. metrics-server) as well
+  as node-level agents (e.g. specific version of CRI). Focus on external or
+  optional services that are needed. For example, if this feature depends on
+  a cloud provider API, or upon an external software-defined storage or network
+  control plane.
+
+  For each of these, fill in the following—thinking about running existing user workloads
+  and creating new ones, as well as about cluster-level services (e.g. DNS):
+  - [Dependency name]
+    - Usage description:
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+
+  Yes and no. This feature will result in additional resource deletion calls, which will
+  scale like the number of pods in the stateful set (ie, one PVC per pod and possibly one
+  PV per PVC depending on the reclaim policy). There will not be additional watches,
+  because the existing pod watches will be used (note: this is to be confirmed in the
+  code).
+  
+  However, anyone who uses this feature would have made those resource deletions anyway:
+  those PVs cost money. So overall there shouldn't be much increase beyond the
+  second-order effect of this feature allowing more automation.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+  PVC deletion may cause PV deletion, depending on reclaim policy, which will result in
+  cloud provider calls through the volume API. However, as noted above, these calls would
+  have been happening anyway, manually.
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+  - PVC, new ownerRef.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by existing SLIs/SLOs?**
+  No. (are there StatefulSet SLOs?)
```

Comment:
not (yet)

Reply:
Could the scale down policy impact pod startup latency in the scale down/up case, since we need to wait for the old PVC to be deleted and then recreated? This is definitely a niche scenario

Reply:
:-) clarified